### ISSUE TITLE
p6doc mentions `Type::*` in instructions / LTA errors on unknown types

### ISSUE THREAD

**Issue by zoffixznet** [perl6/doc#1224]
This is a three-part issue:

1) The instructions of p6doc prepend `Type::` to type names and that [makes users think the `Type::` is a necessary prefix](https://irclog.perlgeek.de/perl6/2017-02-27#i_14173490)

2) If the user looks up a non-existant type, the message is somewhat LTA. I'd expect it to catch the failed use and tell me there's no such type:

```
zoffix@VirtualBox:~$ p6doc -f Blarg.push                                                                                                                                                                                                                       
Could not find Blarg at line 0 in:
    /home/zoffix/.perl6
    /home/zoffix/.rakudobrew/moar-nom/install/share/perl6/site
    /home/zoffix/.rakudobrew/moar-nom/install/share/perl6/vendor
    /home/zoffix/.rakudobrew/moar-nom/install/share/perl6
    CompUnit::Repository::AbsolutePath<68834912>
    CompUnit::Repository::NQP<68268368>
    CompUnit::Repository::Perl5<68268408>
  in sub locate-curli-module at /home/zoffix/.rakudobrew/moar-nom/install/share/perl6/site/resources/C3A18C4AD5209A1B226E53B2C22A5347E5AC4790 line 219
  in sub locate-module at /home/zoffix/.rakudobrew/moar-nom/install/share/perl6/site/resources/C3A18C4AD5209A1B226E53B2C22A5347E5AC4790 line 39
  in sub MAIN at /home/zoffix/.rakudobrew/moar-nom/install/share/perl6/site/resources/C3A18C4AD5209A1B226E53B2C22A5347E5AC4790 line 128
  in block <unit> at /home/zoffix/.rakudobrew/moar-nom/install/share/perl6/site/resources/C3A18C4AD5209A1B226E53B2C22A5347E5AC4790 line 226
```

3) If the user looks up an existent routine but on the wrong type, the utility displays a misleading error: 
``
`p6doc -f Str.push

"No documentation found for method 'push'"
```

**Comment by coke** [comment]
(build not quite right, but closest we have)

**Comment by stmuk** [comment]
All three issues should be addressed as of f75c4997a
